DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -602,12 +607,30 @@ public void UpdateCourseOptions(CourseOptions courseOptions, int customisationId
                     LEFT JOIN dbo.Customisations AS refreshToCu ON refreshToCu.CustomisationID = cu.RefreshToCustomisationId
                     LEFT JOIN dbo.Applications AS refreshToAp ON refreshToAp.ApplicationID = refreshToCu.ApplicationID
                     WHERE
-                        (ap.CourseCategoryID = @categoryId OR @categoryId IS NULL) 
+                        (ap.CourseCategoryID = @categoryId OR @categoryId IS NULL)
                         AND cu.CentreID = @centreId
                         AND ap.ArchivedDate IS NULL
                         AND cu.CustomisationID = @customisationId",
                 new { customisationId, centreId, categoryId }
             ).FirstOrDefault();
         }
+
+        public void SetCompletionDate(int progressId, DateTime? completionDate)
+        {
+            var numberOfAffectedRows = connection.Execute(
+                @"UPDATE Progress
+                        SET Completed = @date
+                        WHERE ProgressID = @progressId",
+                new { date = completionDate, progressId }

```

Comment:
Is there any particular reason we can't just do @completionDate in the query?

Reply:
it didnt seem to like that parameter not being declared

Reply:
You should be able to do
```
var numberOfAffectedRows = connection.Execute(
                @"UPDATE Progress
                        SET Completed = @completionDate
                        WHERE ProgressID = @progressId",
                new { completionDate, progressId }
            );
```

Reply:
Also I've just realised this is in the CourseDataService and has no unit tests. This should really be in the ProgressDataService if we're making a new method. I used the existing method in the CourseDataService for 569 just because it existed.

Reply:
done and added a test in progress data service